src/tty.rs

```diff
@@ -113,6 +113,7 @@ fn openpty(rows: u8, cols: u8) -> (c_int, c_int) {
 /// Really only needed on BSD, but should be fine elsewhere
 fn set_controlling_terminal(fd: c_int) {
     let res = unsafe {
+        #[allow(clippy)]
```

Comment:
What didn't clippy like about this? It would be nice to just disable a single lint rather than all of clippy for this line.

Reply:
This was a problem with multiple platforms. Because `TIOSCTTY` has a different type on macos and linux. So the conversion fails on clippy because it's either an unnecessary conversion, or an incorrect type.

Reply:
I've tried to use `if cfg!(target_os = "macos")` but I wasn't able to make it work. Clippy still complained about the branch that shouldn't be reached on linux.